ReactiveObjC/UITextField+RACCommandSupport.m

```diff
@@ -0,0 +1,56 @@
+//
+//  UITextField+RACCommandSupport.m
+//  ReactiveObjC
+//
+//  Created by Andrew Urban on 9/12/17.
+//  Copyright © 2017 GitHub. All rights reserved.
+//
+
+#import "UITextField+RACCommandSupport.h"
+#import <ReactiveObjC/EXTKeyPathCoding.h>
+#import "RACCommand.h"
+#import "RACDisposable.h"
+#import "RACSignal+Operations.h"
+#import <objc/runtime.h>
+
+static void *UITextFieldRACReturnCommandKey = &UITextFieldRACReturnCommandKey;
+static void *UITextFieldEnabledDisposableKey = &UITextFieldEnabledDisposableKey;
+
+@implementation UITextField (RACCommandSupport)
+
+- (RACCommand *)rac_returnCommand {
+	return objc_getAssociatedObject(self, UITextFieldRACReturnCommandKey);
+}
+
+- (void)setRac_returnCommand:(RACCommand *)command {
+	objc_setAssociatedObject(self, UITextFieldRACReturnCommandKey, command, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
+	
+	// Check for stored signal in order to remove it and add a new one
+	RACDisposable *disposable = objc_getAssociatedObject(self, UITextFieldEnabledDisposableKey);
+	[disposable dispose];
+	
+	if (command == nil) return;
+	
+	disposable = [command.enabled setKeyPath:@keypath(self.enabled) onObject:self];
```

Comment:
I think this needs a `-asScopedDisposable` in order to not leak when the `UITextField` is deallocated.

Reply:
Neither `UIBarButtonItem (RACCommandSupport)` nor `UIButton (RACCommandSupport)` make use of `RACScopedDisposable` in the same case. Do you think all of them should be revised to address this problem?

Reply:
I would think so, but I'm honestly not sure. I'm not very familiar with this code.

Reply:
Me too) So I should investigate more into this possible memory leak but can't do it right now. Will try to look at this soon.